src/core/Akka/IO/Udp.cs

```diff
@@ -34,45 +34,51 @@ public class Udp : ExtensionIdProvider<UdpExt>
     {
         #region internal connection messages
 
-        internal abstract class SocketCompleted : INoSerializationVerificationNeeded { }
+        internal abstract class SocketCompleted : INoSerializationVerificationNeeded
+        {
+            public ByteString Data { get; }
+
+            protected SocketCompleted(SocketAsyncEventArgs eventArgs)
+            {
+                Data = ByteString.CopyFrom(eventArgs.Buffer, eventArgs.Offset, eventArgs.BytesTransferred);
```

Comment:
Instead of embedding the SocketAsyncEventArgs, we extract the data and store that instead.

Reply:
And we make a copy of the data, so there's no problems if the `SocketAsyncEventArgs` get processed and the buffer pool is released.